avocado/core/status/server.py

```diff
@@ -18,17 +20,20 @@ def __init__(self, uri, repo):
         self._server_task = None
 
     async def create_server(self):
+        limit = settings.as_dict().get('nrunner.status_server_buffer_size')
         if ':' in self._uri:
             host, port = self._uri.split(':')
             port = int(port)
             self._server_task = await asyncio.start_server(
                 self.cb,
                 host=host,
-                port=port)
+                port=port,
+                limit=limit)
```

Comment:
Unless I'm missing something, this is not fixing the issue, it is just setting a limit right? What will happen with buffers bigger than this limit?

I'm ok with this if it is a hotfix, but IIUC, another approach is to not have this limit and send/receive buffers in chunks.

Reply:
> Unless I'm missing something, this is not fixing the issue, it is just setting a limit right? What will happen with buffers bigger than this limit?
> 

Yes, this is more of a workaround, but it's one that we *could* raise very very high without side effects.  Like I explained on one commit, there's no allocation of a buffer of that size in advance, so we *could* set a limit as large as you proposed in another commit (8 digits).

> I'm ok with this if it is a hotfix, but IIUC, another approach is to not have this limit and send/receive buffers in chunks.

If you look at the asyncio streams code, you'll find that this already happens in chunks.  The transport can be paused when the buffer gets too "busy".  We may learn that we need better control, but IMO the `readline()` is a higher level abstraction that will work fine here.

On a separate topic, I think we should allow the stdout/stderr data to be (optionally) streamed as it happens, which would reduce the size of the messages.  I think I showed you a PoC that does that.

Reply:
Yes, makes sense.

I have a feeling that `readline()` will not be enough, but since is just a feeling, let's move on. Thanks for the explanation.